### ISSUE TITLE
Higher-level interface to create delegates

### ISSUE THREAD

**Issue by vbo** [servo/core-foundation-rs#205]
It seems currently the only way to create e.g. an NSWindowDelegate is through `objc::declare::ClassDecl` interface:
```rust
let superclass = Class::get("NSObject").unwrap();
let mut decl = ClassDecl::new("MyWindowDelegate", superclass).unwrap();
decl.add_method ...
```
Should this crate expose `trait NSWindowDelegate` and make the creation of such delegate more type-safe?

**Comment by vbo** [comment]
True, this is not about Core Foundation, but it is about Cocoa, which is [a part](https://github.com/servo/core-foundation-rs/blob/master/cocoa/src/appkit.rs) of this crate.

This crate defines `trait NSWindow` for one of the core Cocoa interfaces. The interface provides `setDelegate` method to set `NSWindowDelegate` implementation which controls window behavior. Now currently the binding for this method looks like `setDelegate_(self, delegate: id)` so I can put any object there. I can create this object via `objc` crate, but I was wondering if this is too low-level and if we want to provide type-safe API in this crate to create such a delegate object.

**Comment by nox** [comment]
Oh I see.

**Comment by Litarvan** [comment]
Any update ? I was also using setDelegate and didn't find any other way than the one mentionned by @vbo , may I create a pull request to implement this ?

**Comment by jdm** [comment]
A higher level interface sounds like a useful improvement, yes.

**Comment by Litarvan** [comment]
I tried something in #314
